### PR TITLE
Component nav copy/cut/paste

### PR DESCRIPTION
**Pull request by mjashanks**
#30

### CODE REVIEW

File: packages/builder/src/components/userInterface/ComponentDropdownMenu.svelte [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+}
+
+</script>
+
+<div class="root" on:click|stopPropagation={()  => {}}>
+  <button>
+    <MoreIcon />
+  </button>
+  <ul class="menu"  bind:this={dropdownEl} on:click={hideDropdown}>
+    <li on:click={() => confirmDeleteDialog.show()}>Delete</li>
```
Comment (shogunpurple) [inline packages/builder/src/components/userInterface/ComponentDropdownMenu.svelte:32]: We make use of the `svelte-modal` package now for dialogs - not a huge deal for now but we will be wanting to get rid of anywhere the UIKit modals get used in future

File: packages/builder/src/components/userInterface/ComponentDropdownMenu.svelte [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+const lastPartOfName = c => (c ? last(c._component.split("/")) : "")
+
+const hideDropdown = () => {
+  dropdown.hide()
+}
+
+</script>
+
+<div class="root" on:click|stopPropagation={()  => {}}>
```
Comment (shogunpurple) [inline packages/builder/src/components/userInterface/ComponentDropdownMenu.svelte:27]: What is the empty function for here?

File: packages/builder/src/builderStore/store/index.js [hunk 3]
```diff
@@ -572,10 +576,58 @@ const getPathToComponent = store => component => {
...
+      const parent = getParent(s.currentPreviewItem.props, component._id)
+      parent._children = parent._children.filter(c => c._id !== component._id)
+      _selectComponent(s, parent)
+    }
+
+    return s
+  })
+}
+
+const pasteComponent = store => (targetComponent, mode) => {
```
Comment (shogunpurple) [inline packages/builder/src/builderStore/store/index.js:598]: Having these functions exist at the very top level clutters this already huge file.

It would make more sense to co-locate these functions at the same level at the component tree itself, since that's the only place in the app they are used

**Review by shogunpurple — CHANGES_REQUESTED** [review]
Think we need to move your store functions to the lowest possible level they are used, rather than putting everything in the top level store file. It's very hard to follow sometimes. Otherwise LGTM 🚀

File: packages/builder/src/components/userInterface/ComponentDropdownMenu.svelte (hunk 2 above)
Comment (mjashanks) [reply packages/builder/src/components/userInterface/ComponentDropdownMenu.svelte:27]: oh yes :/ its just to stopPropagation
